duckduckgo/macos-browser
# Turns NetworkProtectionStatusView into a composite view and separates the tunnel view.

Task/Issue URL: https://app.asana.com/0/0/1205247622252423/f
Tech Design URL:
CC:

## Stack PR:

- Feature branch: `diego/feature-network-protection-onboarding`
    - https://github.com/duckduckgo/macos-browser/pull/1465
    - https://github.com/duckduckgo/macos-browser/pull/1466 (this PR)
    - https://github.com/duckduckgo/macos-browser/pull/1474

## Description:

`NetworkProtectionStatusView` has grown to become quite hard to maintain, and the same applies for the view's model.

I need to make some changes to be able to disable specifically the tunnel controller part of the view, and these changes make it a bit easier and cleaner.

This PR splits it into a parent composite view, and a child `TunnelControllerView`.

## Scope:

This PR concentrates refactoring work, meaning that none of the changes should affect functionality in any way.

The only intentional change has been to use the menu item attributes that were missing.  I'll mark this change with a comment.

## Steps to test this PR:

1. Reset all state for Network Protection.
2. Kill the agent app so the new version is loaded (it should be enough to do it through Activity Monitor).
3. Connect NetP, and make sure the flow works correctly.
4. Check that your IP shows up correctly.
5. Ensure the UI in the in-app and menu-app popovers looks as usual.

---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
